src/components/Variables/VariableInternalComment.tsx

```diff
@@ -42,7 +43,7 @@ function VariableInternalComment({ variable, update, editing, sx }: VariableInte
   };
 
   return (
-    (editing || internalComment) && (
+    (editing || internalComment || isDialogOpen) && (
```

Comment:
Without `isDialogOpen` the Dialog closes prematurely (no longer rendered) whenever the text is set to an empty string